.github/workflows/integration_tests.yml

```diff
@@ -749,20 +579,44 @@ jobs:
       - name: Write test report
         id: write_test_report
         if: always()
-        env:
-          TEST_REPORT_FILENAME: test-report-${{ github.run_id }}-${{ matrix.PLATFORM }}-${{ matrix.CLOUD_PROVIDER}}-${{ matrix.PLATFORM_VERSION }}.txt
         run: |
-          echo "write test report to $TEST_REPORT_FILENAME"
-          # please check with headings in print_test_report task roughly 50 lines below
-          echo "| ${{ matrix.PLATFORM }}/${{ matrix.CLOUD_PROVIDER}} ${{ matrix.PLATFORM_VERSION }} | ${{ steps.keptn_install.outcome }} | ${{ steps.authenticate_keptn_cli.outcome }} | ${{ steps.test_airgapped_images.outcome }} | ${{ steps.test_log_ingestion.outcome }} | ${{ steps.test_log_forwarding.outcome }} | ${{ steps.test_sequence_states.outcome }} | ${{ steps.test_sequence_timeout.outcome }} | ${{ steps.test_sequence_control.outcome }} | ${{ steps.test_sequence_queue.outcome }} | ${{ steps.test_sequence_loop.outcome }} | ${{ steps.test_quality_gates.outcome }} | ${{ steps.test_self_healing.outcome }} | ${{ steps.test_delivery_assistant.outcome }} | ${{ steps.test_backup_restore.outcome }} | ${{ steps.test_resource_service.outcome }} | ${{ steps.test_user_managed_delivery.outcome }} | ${{ steps.test_continuous_delivery.outcome }} | ${{ steps.test_continuous_delivery_remote_exec.outcome }} | ${{ steps.test_manage_secrets.outcome }} | ${{ steps.test_webhook_integration.outcome }} | ${{ steps.test_graceful_shutdown.outcome }} | ${{ steps.test_uniform_registration.outcome }} |" > $TEST_REPORT_FILENAME
-          cat $TEST_REPORT_FILENAME
+          cd test/go-tests
+          echo "write test report to ${{ env.FINAL_TEST_REPORT_PATH }}"
+          report_template='{"test":$test_name, "${{ matrix.CLOUD_PROVIDER}}-${{ matrix.PLATFORM_VERSION }}":$test_outcome}'
+
+          test_names=(
+            "${{ matrix.CLOUD_PROVIDER}}-${{ matrix.PLATFORM_VERSION }}/Keptn Install"
+            "${{ matrix.CLOUD_PROVIDER}}-${{ matrix.PLATFORM_VERSION }}/Keptn Auth"
+          )
+
+          test_outcomes=(
+            "${{ steps.keptn_install.outcome }}"
+            "${{ steps.authenticate_keptn_cli.outcome }}"
+          )
+
+          for ((i=0; i<${#test_names[@]}; i++)); do
+            test_name="${test_names[$i]}"
+            test_outcome="${test_outcomes[$i]}"
+          
+            if [[ "$test_outcome" == "" ]]; then
+              test_outcome="skip"
+            elif [[ "$test_outcome" == "success" ]]; then
+              test_outcome="pass"
+            elif [[ "$test_outcome" == "failure" ]]; then
+              test_outcome="fail"
```

Comment:
maybe also add
```
else
   test_outcome="something went wrong"
```

Reply:
The strings I check against in the if/elif are actually constant and coming from GitHub actions ([reference](https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context)) but I will add a panic `else` block just in case :)